### PR TITLE
interfaces/misc: updates for k8s 1.15 (and greengrass test)

### PR DESCRIPTION
**Pull request by jdstrand**
microk8s and upstream k8s has moved to containerd and made various other changes. In updating the policy, discovered that docker-support and kubernetes-support were uses the Getter for UsesPtraceTrace so adjust to use Setter instead and update tests accordingly (adding extra test to greengrass-support in passing).

* interfaces/kubernetes-support: update for k8s 1.15 
* interfaces/docker-support: updates for newer containerd 
* interfaces/docker,kubernetes-support: use SetUsesPtraceTrace()
* interfaces/greengrass-support: adjust testsuite to verify UsesPtraceTrace

### CODE REVIEW

**Comment by jdstrand** [comment]
FYI @ktsakalozos

**Comment by jdstrand** [comment]
@anonymouse64 - fyi re the docker-support updates.

File: interfaces/builtin/docker_support.go [hunk 1]
```diff
@@ -157,7 +172,8 @@ ptrace (read, trace) peer=docker-default,
 # needed by runc for mitigation of CVE-2019-5736
 # For details see https://bugs.launchpad.net/apparmor/+bug/1820344
 / ix,
-/bin/runc rix,
+/bin/runc ixr,
+/pause ixr,
```
Comment (zyga) [inline interfaces/builtin/docker_support.go]: `/pause` is an interesting path. If you know how it happens to work I think a comment would be useful here.

File: interfaces/builtin/docker_support.go [hunk 2]
```diff
@@ -609,7 +625,7 @@ func (iface *dockerSupportInterface) AppArmorConnectedPlug(spec *apparmor.Specif
 	if !release.OnClassic {
 		spec.AddSnippet(dockerSupportConnectedPlugAppArmorCore)
 	}
-	spec.UsesPtraceTrace()
+	spec.SetUsesPtraceTrace()
```
Comment (zyga) [inline interfaces/builtin/docker_support.go]: Ouch, that's a bug!

File: interfaces/builtin/kubernetes_support.go [hunk 3]
```diff
@@ -67,17 +67,23 @@ profile systemd_run (attach_disconnected,mediate_deleted) {
   # Common rules for kubernetes use of systemd_run
   #include <abstractions/base>
 
-  /usr/bin/systemd-run r,
+  /{,usr/}bin/systemd-run rm,
   owner @{PROC}/@{pid}/stat r,
   owner @{PROC}/@{pid}/environ r,
   @{PROC}/cmdline r,
+  @{PROC}/sys/kernel/osrelease r,
```
Comment (zyga) [inline interfaces/builtin/kubernetes_support.go]: The surface of kernel interfaces is daunting.

File: interfaces/builtin/kubernetes_support.go [hunk 4]
```diff
@@ -197,14 +203,14 @@ func (iface *kubernetesSupportInterface) AppArmorConnectedPlug(spec *apparmor.Sp
 	case "kubelet":
 		systemd_run_extra = kubernetesSupportConnectedPlugAppArmorKubeletSystemdRun
 		snippet += kubernetesSupportConnectedPlugAppArmorKubelet
-		spec.UsesPtraceTrace()
+		spec.SetUsesPtraceTrace()
```
Comment (zyga) [inline interfaces/builtin/kubernetes_support.go]: More bugs?

File: interfaces/builtin/kubernetes_support.go [hunk 5]
```diff
@@ -197,14 +203,14 @@ func (iface *kubernetesSupportInterface) AppArmorConnectedPlug(spec *apparmor.Sp
...
 	case "kubeproxy":
 		snippet += kubernetesSupportConnectedPlugAppArmorKubeproxy
 	default:
 		systemd_run_extra = kubernetesSupportConnectedPlugAppArmorKubeletSystemdRun
 		snippet += kubernetesSupportConnectedPlugAppArmorKubelet
 		snippet += kubernetesSupportConnectedPlugAppArmorKubeproxy
-		spec.UsesPtraceTrace()
+		spec.SetUsesPtraceTrace()
```
Comment (zyga) [inline interfaces/builtin/kubernetes_support.go]: And here?

File: interfaces/builtin/kubernetes_support.go [hunk 6]
```diff
@@ -67,17 +67,23 @@ profile systemd_run (attach_disconnected,mediate_deleted) {
...
 
+  # ptrace 'trace' is coarse and not required for using the systemd private
+  # socket, and while the child profile omits 'capability sys_ptrace', skip
+  # for now since it isn't strictly required.
+  deny ptrace trace peer=unconfined,
   /run/systemd/private rw,
-  /bin/true ixr,
-  ptrace trace peer=unconfined,
+
+  /{,usr/}bin/true ixr,
```
Comment (anonymouse64) [inline interfaces/builtin/kubernetes_support.go]: not a blocker, but out of curiosity, why is this in the policy in kubernetes-support to begin with? shouldn't something as common as  `/bin/true` be in the base apparmor template?

Also FWIW, /bin/true exists in both core and core18, so we could add it to the base template I think

File: interfaces/builtin/docker_support.go (hunk 2 above)
Comment (jdstrand) [reply interfaces/builtin/docker_support.go]: Yep, here and below. I recall that there was a change in the API for the Getter and the Setter during the review. Guess we missed it :\

File: interfaces/builtin/kubernetes_support.go (hunk 6 above)
Comment (jdstrand) [reply interfaces/builtin/kubernetes_support.go]: @anonymouse64 - it is already in the base template. This is in the systemd_run which the child profile that doesn't use that template. The apparmor base abstraction could conceivably use it, as you mention, but we would need to carry it here regardless until everyone upgraded to the new apparmor.

File: interfaces/builtin/kubernetes_support.go (hunk 4 above)
Comment (jdstrand) [reply interfaces/builtin/kubernetes_support.go]: Yep

File: interfaces/builtin/kubernetes_support.go (hunk 5 above)
Comment (jdstrand) [reply interfaces/builtin/kubernetes_support.go]: Yep

**Pull request merged**